parquet-column/src/main/java/org/apache/parquet/column/statistics/geometry/EnvelopeCovering.java

```diff
@@ -0,0 +1,225 @@
+  // The POC only supports EPSG:3857 and EPSG:4326 at the moment
+  private static final List<String> SUPPORTED_CRS = Arrays.asList("EPSG:3857", "EPSG:4326");
+
+  private static final ByteBuffer EMPTY = ByteBuffer.wrap(new byte[0]);
+  private final WKBReader reader = new WKBReader();
+  private final WKBWriter writer = new WKBWriter();
+  private final GeometryFactory factory = new GeometryFactory();
+  private LogicalTypeAnnotation.Edges edges;
+  private String crs;
+
+  public EnvelopeCovering(LogicalTypeAnnotation.Edges edges, String crs) {
+    super(EMPTY, DEFAULT_COVERING_KIND);
+    this.edges = edges;
+    this.crs = crs;
+    validateSupportedCrs(this.crs);
+    validateKind(this.kind);
+  }
+
+  public EnvelopeCovering() {
+    super(EMPTY, DEFAULT_COVERING_KIND);
+  }
+
+  private static void validateKind(String kind) {
+    Preconditions.checkArgument(kind.equalsIgnoreCase(DEFAULT_COVERING_KIND), "kind only accepts WKB");
+  }
+
+  private static void validateSupportedCrs(String crs) {
+    if (!SUPPORTED_CRS.contains(crs)) {
+      LOG.warn("Unsupported CRS: {}. Supported CRS are EPSG:3857 and EPSG:4326.", crs);
+    }
+  }
+
+  @Override
+  public String getKind() {
+    return kind + "|" + crs + "|" + edges;
+  }
+
+  @Override
+  public void setKind(String kind) {
+    if (kind == null || kind.isEmpty()) {
+      throw new IllegalArgumentException("Kind cannot be null or empty");
+    }
+
+    // Split the input string by the "|" delimiter
+    String[] parts = kind.split("\\|");
+
+    // Ensure we have exactly 3 parts: kind, crs, and edges
+    if (parts.length != 3) {
+      throw new IllegalArgumentException("Invalid kind format. Expected format: 'kind|crs|edges'");
+    }
+
+    // Assign the values to the respective fields
+    this.kind = parts[0];
+    this.crs = parts[1];
+
+    try {
+      this.edges = LogicalTypeAnnotation.Edges.valueOf(parts[2]);
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException("Invalid edges value: " + parts[2], e);
+    }
+  }
+
+  @Override
+  void update(Geometry geom) {
+    if (geom == null) {
+      return;
+    }
+    try {
+      if (value != EMPTY) {
+        Geometry existingGeometry = reader.read(value.array());
+        Envelope existingEnvelope = createEnvelopeFromPolygon(existingGeometry);
+        Envelope newEnvelope = geom.getEnvelopeInternal();
+
+        Envelope combinedEnvelope = edges == LogicalTypeAnnotation.Edges.SPHERICAL
+            ? extendEnvelopeSpherical(existingEnvelope, newEnvelope)
+            : extendEnvelopePlanar(existingEnvelope, newEnvelope);
+
+        Geometry envelopePolygon = createPolygonFromEnvelope(combinedEnvelope);
+
+        value = ByteBuffer.wrap(writer.write(envelopePolygon));
+      } else {
+        Geometry envelopePolygon = createPolygonFromEnvelope(geom.getEnvelopeInternal());
+        value = ByteBuffer.wrap(writer.write(envelopePolygon));
+      }
+    } catch (ParseException e) {
+      value = null;
```

Comment:
Add an error log for what has happened?

Reply:
BYW, should we catch all exceptions here? We don't want to fail the writing if anything wrong happens here, e.g. `extendEnvelopeSpherical` might throw for spherical features.